src/index.ts

```diff
@@ -82,19 +82,19 @@ const remarkEmbedder: Plugin<[RemarkEmbedderOptions]> = ({
       }
 
       const {children} = paragraphNode
-      const node = children[0] as Literal & Parent
+      const node = children[0] as Link | Text
       const isText = node.type === 'text'
       // it's a valid link if there's no title, and the value is the same as the URL
       const isValidLink =
         node.type === 'link' &&
-        node.title === null &&
+        node.title === undefined &&
```

Comment:
According to `@types/mdast`, `title` is of type `string | undefined` and not `null`

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f2b0b6ec0095de2c52b9799bb3260ee17f79182b/types/mdast/index.d.ts#L201-L204

Reply:
@wooorm this seems not to be correct according to our tests 🤔

Reply:
You could use both. Some tools use one or the other

Reply:
So types need to be changed then? 🤔

As using `undefined` is breaking our tests now

Reply:
Could just change it to `node.title == null`.

Reply:
I’d use `!node.title` as the comment above this conditions read “no title”.

`null` has been historically used everywhere (by me). TypeScript folks seem to not like `null`, hence type definitions only uses `undefined` 🤷‍♂️ We had some discussions about it if I remember correctly. I’m slowly moving towards `undefined`, but IMO most code should check for both in most cases.

Reply:
Will change to double equals instead of triple equals

Will also create a PR to update the types, to make them correct.